lib/src/robot/sessions_client.dart

```diff
@@ -65,6 +65,7 @@ class SessionsClient implements ResourceRPCClient {
       });
     } catch (e) {
       _logger.e('error starting session: $e');
+      reset();
```

Comment:
is this where you start a new session upon failing? and what does failing mean in this context? same with starting a new session?

Reply:
I added a call to `metadata()` inside reset(), that is where it starts a new session upon failing.